src/FSharp.Core/mailbox.fs

```diff
@@ -175,8 +176,9 @@ type Mailbox<'Msg>(cancellationSupported: bool) =
     member x.Post msg =
         lock syncRoot (fun () ->
 
-            // Add the message to the arrivals queue
-            arrivals.Enqueue msg
+            if hasNotDisposed then
```

Comment:
I have a bad feeling about silently ignoring the message altogether.
Wouldn't a `ObjectDisposedException` be more appropriate (considering it's use within .NET)